Add new methods

tripledoublev opened the pull request:
- list_files_from_hash
- get_file_from_collection_hash

tripledoublev committed:
Add list_files_from_hash method

tripledoublev committed:
Create get_file_from_collection_hash

RangerMauve commented on src/iroh.rs:
```diff
@@ -471,6 +471,51 @@ impl VeilidIrohBlobs {
+        path: &str,
+    ) -> Result<Hash> {
+        // Fetch the collection entry from the store using the collection hash
+        let entry = self
```
these should be put in a method like `get_collection_from_hash to reuse. get_collection should also use this new method

tripledoublev replied:
7e9cffd7c67b15c7eade7fd003136eca28ff7bfc
Refactors the list_files_from_hash method. It moves code inside a new method called get_collection_from_hash. The get_collection_from_hash method retrieves the collection HashMap using the collection hash.

tripledoublev replied:
get_collection currently gets the collection from the name.

if it uses get_collection_from_hash, the following methods will need to be refactored to take the hash instead of the name:
- set_file
- get_file
- delete_file
- list_files

should i refactor to make these use the collection_hash instead of collection_name?


current implementation:
`set_file(collectionName, path: string, hash) -> Result<Hash>` (return collection hash)
`get_file(collectionName, path) -> Result<Hash>`
`delete_file(collectionNAme, path) -> Result<Hash> `(return collection hash)
`list_files(collectionNAme)  -> Result`<Vec<string>>

RangerMauve reviewed (changes requested):
Looks great, just minor code reuse comments

tripledoublev committed:
Refactor list_files_from_hash method and add get_collection_from_hash method

tripledoublev committed:
add set_file_from_collection_hash method

tripledoublev committed:
add delete_file_from_collection_hash method

tripledoublev committed:
Refactor update_collection to take name, hash and collection

tripledoublev committed:
Add helper functions update_collection_with_name and update_collection_with_hash

tripledoublev committed:
Add get_name_from_hash method to retrieve collection name from hash

tripledoublev committed:
Refactor update_collection to take name, hash and collection

tripledoublev committed:
Adjust calls to update_collection

tripledoublev commented:
Went ahead with the creation of new methods and some light refactoring.
This should be able to cover all scenarios. We can remove and/or rename to streamline it all.

**New Methods:**
- `set_file_from_collection_hash`: Added a new method to set files in a collection using the collection_hash instead of collection_name.
- `delete_file_from_collection_hash`: Added a new method to delete files from a collection using the collection_hash.

Refactor:
- `update_collection`: Refactored the `update_collection `method to take either a `collection_name `or `collection_hash `and the collection itself. This allows for more flexible operations when managing collections.

Helper Functions:
- `update_collection_with_name`: Added a helper function that simplifies updating collections by name.
- `update_collection_with_hash`: Added a helper function to update collections directly by their hash.

New Helper Method:
- `get_name_from_hash`: Added a method to retrieve the collection_name from a collection_hash, which is used in cases where only the hash is available.

Adjusted Calls:
- Updated existing calls to update_collection to ensure they accommodate the refactored method and its flexible signature.

tripledoublev committed:
Refactor update_collection to delegate to update_collection_with_hash

tripledoublev committed:
Adjust update_collection call to use update_collection_with_name

tripledoublev committed:
Refactor store_tag to persist_collection_with_name
